v2/tools/generator/internal/codegen/pipeline/flatten_resources.go

```diff
@@ -54,7 +54,11 @@ func FlattenResources() *Stage {
 		})
 }
 
-func flattenResource(defs astmodel.TypeDefinitionSet, t astmodel.Type) (astmodel.Type, error) {
+func flattenResource(defs astmodel.TypeDefinitionSet, t astmodel.Type, depth int) (astmodel.Type, error) {
+	if depth > 10 {
```

Comment:
This sounds like you had a nasty recursion bug and then got some very nasty debugging output!

Reply:
Yup!